### PR TITLE
crucible-jvm: Add --java-bin-dirs

### LINKED ISSUE #633: crucible-jvm: Allow specifying Java's location with --java-bin-dirs/PATH
Currently, the `crucible-jvm` executable allows specifying the locations of Java classes through the `-j` flag (for JAR files) and the `-c` flag (for `.class` files). This has two issues:

1. For JDK 8 and earlier, this requires finding where your JDK installation's `rt.jar` file is and manually passing it to `crucible-jvm`. This is tedious.
2. This approach doesn't work at all for JDK 9 and later—see GaloisInc/saw-script#861.

Elsewhere, SAW aims to fix these issues with GaloisInc/saw-script#1030, which adds a `-b` flag which points to the location where the `java` executable lives. (Alternatively, if `-b` isn't set, SAW can also find `java` if it is located on the `PATH`.) With this information, one doesn't need to specify where `rt.jar` lives at all, as it can be inferred from `java -XshowSettings:properties`. Moreover, knowing where `java` is located is essential to getting JDK 9 and later to work.

I propose we adopt a similar `-b`/`PATH` scheme for `crucible-jvm`.

### PR DESCRIPTION
**Pull request by RyanGlScott**
This mirrors a similar recent change introduced to SAW in GaloisInc/saw-script#1030.

While I was in town, I updated the two `crucible-jvm`–related test suites to use the `PATH` instead of the `-j` flag. I can't exactly confirm via CI if these changes work (see #634), but they appear to do the right thing locally.

Fixes #633.

### CODE REVIEW

File: crucible-jvm/tool/Main.hs [hunk 1]
```diff
@@ -129,6 +158,11 @@ cruxJVMConfig = Crux.Config
         $ Crux.ReqArg "TODO"
         $ \p opts ->
             Right $ opts { jarList = jarList opts ++ splitSearchPath p }
+      , Crux.Option ['b'] ["java-bin-dirs"]
+        "TODO"
```
Comment (atomb) [inline crucible-jvm/tool/Main.hs:162]: It would be nice to have a real description for this option.

File: crucible-jvm/tool/Main.hs [hunk 2]
```diff
@@ -129,6 +158,11 @@ cruxJVMConfig = Crux.Config
...
+        $ Crux.ReqArg "TODO"
```
Comment (atomb) [inline crucible-jvm/tool/Main.hs:163]: Likewise here.

**Review by atomb — APPROVED** [review]
This looks good to me, though I think real descriptions for the new (and old!) options in `crucible-jvm` might be worthwhile. I hope that once #635 is merged, and this is updated to include that, the CI problems will go away.

File: crucible-jvm/tool/Main.hs (hunk 1 above)
Comment (RyanGlScott) [reply crucible-jvm/tool/Main.hs]: Fair enough. I copied over this `TODO` from the existing `TODO`s above, but it's easy enough to just fill them in. I've done so in a commit that I just pushed.

**Pull request merged**